kratos/python_scripts/response_functions/response_function_interface.py

```diff
@@ -1,19 +1,8 @@
-
-# ==============================================================================
-#  KratosShapeOptimizationApplication
-#
-#  License:         BSD License
-#                   license: ShapeOptimizationApplication/license.txt
-#
-#  Main authors:    Geiser Armin, https://github.com/armingeiser
-#
-# ==============================================================================
-
-class ResponseFunctionBase(object):
-    """The base class for response functions. Each response function
+class ResponseFunctionInterface(object):
```

Comment:
This is the new file in the core - Github shows it as moved...

Reply:
Just out of curiosity: Is there a reason, why you called it "Interface" rather than "Base" or just "ResponseFunction"?

Reply:
The [AdjointResponseFunction](https://github.com/KratosMultiphysics/Kratos/blob/master/kratos/response_functions/adjoint_response_function.h) provides some of the necessary components for an adjoint sensitivity analysis:

* gradient w.r.t to the primal variables (and their derivatives)
* partial gradient w.r.t to the design variable

However it does **not** calculate the total gradient of the response function w.r.t the design variables.
This can only be obtained using the whole framework of adjoint_solver, adjoint_response_function and sensitivity_builder.

Besides the adjoint approach to calculate the gradient, there are several other ways - fore some e.g. geometrical response functions, the adjoint approach does not even make sense. 

This response function interface provides a unique interface for all possible ways to calculate the value and gradients of a response. 
This is necessary because e.g. in optimization i do not care about the implementation details (e.g. how the adjoint sensitivity analysis uses the adjoint_response_function.h ...) but i am just interested in the value and gradient of the response.

Reply:
cool thx for the explanation

Could you please add sth like this to the docstring of the class?